static/components/TaskDetails.jsx

```diff
@@ -58,29 +62,35 @@ var TaskDetails = React.createClass({
   componentDidUpdate(prevProps, prevState) {
     let task = this.state.task
 
-    if(prevState.task !== null && task.status === 'finished' && prevState.task.status === 'in_progress'){
-      let {name}                = task.app
-      let {environment, number} = task
-      let path                  = `/deploys/${name}/${environment}/${number}`
+    if(task){
```

Comment:
In one of my previous reviews with ben, he suggests to avoid deep nesting 
https://github.com/getsentry/sentry/pull/4931#discussion_r130231722
https://github.com/getsentry/sentry/pull/4931#discussion_r130232188

I would agree, and in this case would improve the diff for reviewers.